pkg/pool-weighted/test/foundry/E2eSwapBigDecimalsToSmall.t.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import "forge-std/Test.sol";
+
+import { ERC20TestToken } from "@balancer-labs/v3-solidity-utils/contracts/test/ERC20TestToken.sol";
+
+import { E2eSwapWeightedTest } from "./E2eSwap.t.sol";
+
+contract E2eSwapWeightedBigDecimalsToSmallTest is E2eSwapWeightedTest {
+    function setUpTokens() internal override {
+        tokenA = new ERC20TestToken("tokenA", "tokenA", 12);
```

Comment:
Are the decimal choices deliberate? Here it is 6 & 12; elsewhere we have 8 & 14.

Should it work with the full range of decimals? We'd talked about putting in the docs that although we technically support even 0-decimal tokens (and those silly 2-decimal stable coins), those aren't going to work well in an AMM.

The lowest decimal token in common use would be USDC and friends, with 6, as you have it here. (And then WBTC & Co. with 8 decimals.) We can say we don't recommend using < 6 decimals; do so at your own risk.

Should this test from 6 to 18 then? (Could also try < 6, just to see if/where it breaks, even if we leave it at 6 in the end.)

Could we have a test that loops through 6 to 17, testing them all against 18? (Probably overkill to do all the permutations.) You'd have to refactor a little to call the token setup again